### PR TITLE
Provide additional data on deployment

### LINKED ISSUE #1682: Implement additional parameter for contract deployment
See neo-project/neo#2239.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1682 .

Also follow https://github.com/neo-project/neo/pull/2241/ .

### CODE REVIEW

File: pkg/core/interop/context.go [hunk 1]
```diff
@@ -135,7 +135,16 @@ func NewContractMD(name string, id int32) *ContractMD {
...
+	index := sort.Search(len(c.Manifest.ABI.Methods), func(i int) bool {
+		md := c.Manifest.ABI.Methods[i]
+		if md.Name != desc.Name {
+			return md.Name >= desc.Name
+		}
+		return len(md.Parameters) > len(desc.Parameters)
+	})
+	c.Manifest.ABI.Methods = append(c.Manifest.ABI.Methods, manifest.Method{})
+	copy(c.Manifest.ABI.Methods[index+1:], c.Manifest.ABI.Methods[index:])
+	c.Manifest.ABI.Methods[index] = *desc
```
Comment (AnnaShaleva) [inline pkg/core/interop/context.go:147]: Set `desc.Safe` firstly (it is done 2 lines below), and after that do `c.Manifest.ABI.Methods[index] = *desc`.

File: cli/testdata/deploy/main.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ var key = "key"
 
 const mgmtKey = "mgmt"
 
-func _deploy(isUpdate bool) {
+func _deploy(data interface{}, isUpdate bool) {
```
Comment (AnnaShaleva) [inline cli/testdata/deploy/main.go:16]: We need to change all our examples (`neo-go/examples`) because some of them also have `deploy` method.

**Review by AnnaShaleva — CHANGES_REQUESTED** [review]

File: pkg/core/interop/context.go (hunk 1 above)
Comment (fyrchik) [reply pkg/core/interop/context.go:147]: Fixed, actually this bug was already present.

File: cli/testdata/deploy/main.go (hunk 2 above)
Comment (fyrchik) [reply cli/testdata/deploy/main.go:16]: Fixed.

**Review by AnnaShaleva — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**